src/gt4py/next/iterator/embedded.py

```diff
@@ -113,6 +121,44 @@ def mapped_index(
         assert common.is_int_index(res)
         return res
 
+    if dace:
+        # Extension of NeighborTableOffsetProvider adding SDFGConvertible support in GT4Py Programs
+        def data_ptr(self) -> int:
+            obj = self.table
+            if dace.dtypes.is_array(obj) and (
+                hasattr(obj, "__array_interface__") or hasattr(obj, "__cuda_array_interface__")
+            ):
+                if dace.dtypes.is_gpu_array(obj):
+                    return obj.__cuda_array_interface__["data"][0]  # type: ignore
+                else:
+                    return obj.__array_interface__["data"][0]  # type: ignore
+            else:
+                raise ValueError("Unsupported data container.")
+
+        def __descriptor__(self) -> dace.data.Data:
+            return dace.data.create_datadescriptor(self.table)
+
+
+class NeighborOffsetProvider:
+    def __init__(
+        self,
+        table: core_defs.NDArrayObject,
+        origin_axis: common.Dimension,
+        neighbor_axis: common.Dimension,
+        max_neighbors: int,
+        has_skip_values=True,
+    ) -> None:
+        self.table = None
+        self.origin_axis = origin_axis
+        self.neighbor_axis = neighbor_axis
+        self.max_neighbors = max_neighbors
+        self.has_skip_values = has_skip_values
+        self.index_type = table.dtype
+
+    def mapped_index(
+        self, cur_index: int | np.integer, neigh_index: int | np.integer
+    ) -> Optional[int | np.integer]: ...
+
```

Comment:
The preliminary name for mine is `CompileConnectivity`. I think `CompileTimeConnectivity` would be even better. I have been using the same `Compile` prefix for compile-time standins for field arguments as well. I could also imagine `Static` as a prefix instead.

Reply:
FYI, I have named it `CompileTimeConnectivity`. As decided with @DropD, since this PR will be merged first, we leave it as is and then Rico removes my `CompileTimeConnectivity` class from embedded.py (TODO in his PR).